docs/ops/deployment/docker.md

```diff
@@ -24,119 +24,476 @@ under the License.
+the same way as in the `flink-conf.yaml`.
+The entry point script will override or add the listed options to the `flink-conf.yaml` before starting the Flink process in the container.
+
+### Provide custom configuration
+
+The configuration files (`flink-conf.yaml`, logging, hosts etc) are located in the `/opt/flink/conf` directory in the Flink image.
+To provide a custom location for the Flink configuration files, you can
+
+* **either mount a volume** with the custom configuration files to this path `/opt/flink/conf` when you run the Flink image:
+
+    ```sh
+    docker run \
+        --mount type=bind,src=/host/path/to/custom/conf,target=/opt/flink/conf \
+        flink:{% if site.is_stable %}{{site.version}}-scala{{site.scala_version_suffix}}{% else %}latest{% endif %} <jobmanager|standalone-job|taskmanager>
+    ```
+
+* or add them to your **custom Flink image**, build and run it:
+
+    ```dockerfile
+    FROM flink
+    ADD /host/path/to/flink-conf.yaml /opt/flink/conf/flink-conf.yaml
+    ADD /host/path/to/log4j.properties /opt/flink/conf/log4j.properties
+    ```
+
+<span class="label label-warning">Warning!</span> The mounted volume must contain all necessary configuration files.
+The `flink-conf.yaml` file must have write permission so that the docker entry point script can modify it in certain cases.
+
+### Using plugins
 
-## Using plugins
 As described in the [plugins]({{ site.baseurl }}/ops/plugins.html) documentation page: in order to use plugins they must be
 copied to the correct location in the Flink installation for them to work.
 
-When running Flink from one of the provided Docker images by default no plugins have been activated.
-The simplest way to enable plugins is to modify the provided official Flink docker images by adding
-an additional layer. This does however assume you have a docker registry available where you can push images to and
-that is accessible by your cluster.
+If you want to enable some optional plugins provided with Flink, you can just pass an environment variable `ENABLE_BUILT_IN_PLUGINS`
+when you run the Flink image.
+The `ENABLE_BUILT_IN_PLUGINS` should contain a list of plugin jar file names separated by `;`.
 
-As an example assume you want to enable the [S3]({{ site.baseurl }}/ops/filesystems/s3.html) plugins in your installation.
+    ```sh
+    docker run \
+        --env ENABLE_BUILT_IN_PLUGINS=flink-plugin1.jar;flink-plugin2.jar \
+        flink:{% if site.is_stable %}{{site.version}}-scala{{site.scala_version_suffix}}{% else %}latest{% endif %} <jobmanager|standalone-job|taskmanager>
+    ```
 
-Create a Dockerfile with a content something like this: {% highlight dockerfile %}
-# On which specific version of Flink is this based?
-# Check https://hub.docker.com/_/flink?tab=tags for current options
-FROM flink:{{ site.version }}-scala_2.12
+Otherwise you can use other methods, described [here](#advanced-customization).
```

Comment:
I think it does hurt to improve linking parts of the docs. The chapters can get reordered and user might just skim certain chapters missing the bigger picture and links attract more attention.

I will though change to:
`Otherwise you can use other [advanced methods](#advanced-customization).`

Reply:
Ok, if it sounds better, just when adding links, I am trying to follow [the last best practice section of links code style in docs](https://flink.apache.org/contributing/docs-style.html#links), so I can change it to:
`There are also more [advanced ways](#advanced-customization) for customizing the Flink image.`

Reply:
sounds good.